sys.objects.lua

```diff
@@ -378,15 +378,17 @@ init_tech_objects = function()
         --Группа сигналов, по наличию которых автоматически включается шаг.
         if value.enable_step_by_signal then
             for sub_group, item in pairs( value.enable_step_by_signal ) do
-                if type( item ) == "boolean" then
-                    local step_on = mode[ state_n ][ step_n ][ step.A_ENABLE_STEP_BY_SIGNAL ]
-                    --Добавляем параметр отключения/не отключения шага при
-                    --пропадании сигналов.
-                    step_on:set_bool_property( item )
-
-                elseif type( item ) == "table" then
-                    process_dev_ex( mode, state_n, step_n, step.A_ENABLE_STEP_BY_SIGNAL,
-                        item, 0, sub_group - 1 )
+                for _, sub_item in pairs( item ) do

```

Comment:
```suggestion
                for _, sub_item in pairs( value.enable_step_by_signal ) do

```
В данной реализации происходит следующее:

```
 Lua error: ./sys/sys.objects.lua:381: bad argument #1 to 'pairs' (table expected, got boolean)
2022-07-21 08.25.39 ERROR  (3) -> Lua stack traceback:
        [C]:-1: in function 'pairs' ('global')
        ./sys/sys.objects.lua:381: in function 'process_step' ('local')
        ./sys/sys.objects.lua:575: in function '(null)'
('')
```

Reply:
@alexandraneroda вы со мной согласны?